### ISSUE TITLE
Calling #inspect on an Event instance doesn't return the content of an ArrayList.

### ISSUE THREAD

**Issue by ph** [elastic/logstash#3308]
Calling #inspect on a event which contains an ArrayList will return the instance id instead of returning a string of the actual content.

``
"instances"=>#<Java::JavaUtil::ArrayList:0x205196d9>
``

**Comment by andrewvc** [comment]
@ph what are your thoughts on simply overriding `#inspect` for Event and simply invoking #to_json ? The output will be like:

```json
{"@timestamp":"2013-01-01T00:00:00.000Z","type":"sprintf","message":"hello world","tags":["tag1"],"source":"/home/foo","a":"b","c":{"d":"f","e":{"f":"g"}},"f":{"g":{"h":"i"}},"j":{"k1":"v","k2":["w","x"],"k3":{"4":"m"},"k4":[{"nested":"cool"}],"5":7},"nilfield":null,"@version":"1"}
```

vs. 

```
#<LogStash::Event:0x4a41caed @metadata_accessors=#<LogStash::Util::Accessors:0x6b4fd7d @store={"fancy"=>"pants", "have-to-go"=>{"deeper"=>"inception"}}, @lut={}>, @cancelled=false, @data={"@timestamp"=>"2013-01-01T00:00:00.000Z", "type"=>"sprintf", "message"=>"hello world", "tags"=>#<Java::JavaUtil::ArrayList:0x26f480c6>, "source"=>"/home/foo", "a"=>"b", "c"=>{"d"=>"f", "e"=>{"f"=>"g"}}, "f"=>{"g"=>{"h"=>"i"}}, "j"=>{"k1"=>"v", "k2"=>#<Java::JavaUtil::ArrayList:0x615c4ea4>, "k3"=>{"4"=>"m"}, "k4"=>#<Java::JavaUtil::ArrayList:0x1dd76982>, "5"=>7}, "nilfield"=>nil, "@version"=>"1"}, @metadata={"fancy"=>"pants", "have-to-go"=>{"deeper"=>"inception"}}, @accessors=#<LogStash::Util::Accessors:0x72237be8 @store={"@timestamp"=>"2013-01-01T00:00:00.000Z", "type"=>"sprintf", "message"=>"hello world", "tags"=>#<Java::JavaUtil::ArrayList:0x26f480c6>, "source"=>"/home/foo", "a"=>"b", "c"=>{"d"=>"f", "e"=>{"f"=>"g"}}, "f"=>{"g"=>{"h"=>"i"}}, "j"=>{"k1"=>"v", "k2"=>#<Java::JavaUtil::ArrayList:0x615c4ea4>, "k3"=>{"4"=>"m"}, "k4"=>#<Java::JavaUtil::ArrayList:0x1dd76982>, "5"=>7}, "nilfield"=>nil, "@version"=>"1"}, @lut={"host"=>[{"@timestamp"=>"2013-01-01T00:00:00.000Z", "type"=>"sprintf", "message"=>"hello world", "tags"=>#<Java::JavaUtil::ArrayList:0x26f480c6>, "source"=>"/home/foo", "a"=>"b", "c"=>{"d"=>"f", "e"=>{"f"=>"g"}}, "f"=>{"g"=>{"h"=>"i"}}, "j"=>{"k1"=>"v", "k2"=>#<Java::JavaUtil::ArrayList:0x615c4ea4>, "k3"=>{"4"=>"m"}, "k4"=>#<Java::JavaUtil::ArrayList:0x1dd76982>, "5"=>7}, "nilfield"=>nil, "@version"=>"1"}, "host"], "message"=>[{"@timestamp"=>"2013-01-01T00:00:00.000Z", "type"=>"sprintf", "message"=>"hello world", "tags"=>#<Java::JavaUtil::ArrayList:0x26f480c6>, "source"=>"/home/foo", "a"=>"b", "c"=>{"d"=>"f", "e"=>{"f"=>"g"}}, "f"=>{"g"=>{"h"=>"i"}}, "j"=>{"k1"=>"v", "k2"=>#<Java::JavaUtil::ArrayList:0x615c4ea4>, "k3"=>{"4"=>"m"}, "k4"=>#<Java::JavaUtil::ArrayList:0x1dd76982>, "5"=>7}, "nilfield"=>nil, "@version"=>"1"}, "message"]}>>
```

**Comment by ph** [comment]
@andrewvc I'm up for calling `to_json` from `#inspect` since it is mostly only used for debugging and we we don't have any other variables that should be included other than `@data`, make sure you use  `#to_json_with_metadata` and not `#to_json` so have include all the data in the inspect.

**Comment by andrewvc** [comment]
@ph this is now fixed in this new PR: https://github.com/elastic/logstash/pull/3474

**Comment by andrewvc** [comment]
Fixed in #3474
